File: rename-components.mjs
import fs from 'fs';

// Read the file
const content = fs.readFileSync('frontend/src/pages/StrategicDashboard.jsx', 'utf-8');

// Replace component names
let newContent = content
  // Replace styled component declarations (lines ~1999+)
  .replace(/^(1999:)(const CompetitorCard = styled\.div`)/m, '$1const CompetitorKeywordCard = styled.div`')
  .replace(/^(2012:)(const CompetitorCardHeader = styled\.div`)/m, '$1const CompetitorKeywordCardHeader = styled.div`')
  .replace(/^(2019:)(const CompetitorName = styled\.h4`)/m, '$1const CompetitorKeywordName = styled.h4`')
  .replace(/^(2025:)(const CompetitorBadge = styled\.span`)/m, '$1const CompetitorKeywordBadge = styled.span`')
  .replace(/^(2038:)(const CompetitorMetric = styled\.div`)/m, '$1const CompetitorKeywordMetric = styled.div`')
  .replace(/^(2049:)(const CompetitorMetricLabel = styled\.span`)/m, '$1const CompetitorKeywordMetricLabel = styled.span`')
  .replace(/^(2054:)(const CompetitorMetricValue = styled\.span`)/m, '$1const CompetitorKeywordMetricValue = styled.span`')

  // Replace JSX usage in the competitor monitoring section only
  .replace(/<CompetitorCard key={index}/g, '<CompetitorKeywordCard key={index')
  .replace(/<\/CompetitorCard>/g, '</CompetitorKeywordCard>')
  .replace(/<CompetitorCardHeader>/g, '<CompetitorKeywordCardHeader>')
  .replace(/<\/CompetitorCardHeader>/g, '</CompetitorKeywordCardHeader>')
  .replace(/\{competitor\.competitorAppName\}<\/CompetitorName>/g, '{competitor.competitorAppName}</CompetitorKeywordName>')
  .replace(/<CompetitorName>\{competitor\.competitorAppName\}/g, '<CompetitorKeywordName>{competitor.competitorAppName}')
  .replace(/<CompetitorBadge priority/g, '<CompetitorKeywordBadge priority')
  .replace(/<\/CompetitorBadge>/g, '</CompetitorKeywordBadge>')
  .replace(/<CompetitorMetric>/g, '<CompetitorKeywordMetric>')
  .replace(/<\/CompetitorMetric>/g, '</CompetitorKeywordMetric>')
  .replace(/<CompetitorMetricLabel>/g, '<CompetitorKeywordMetricLabel>')
  .replace(/<\/CompetitorMetricLabel>/g, '</CompetitorKeywordMetricLabel>')
  .replace(/<CompetitorMetricValue value/g, '<CompetitorKeywordMetricValue value')
  .replace(/<\/CompetitorMetricValue>/g, '</CompetitorKeywordMetricValue>');

// Write back
fs.writeFileSync('frontend/src/pages/StrategicDashboard.jsx', newContent);

console.log('Components renamed successfully!');

File: rename-components.py
#!/usr/bin/env python3
import re

# Read the file
with open('frontend/src/pages/StrategicDashboard.jsx', 'r', encoding='utf-8') as f:
    content = f.read()

# Rename the competitor keyword monitoring components (only the ones I added starting around line 1999)
replacements = [
    # Component declarations (my new components only)
    (r'(1999:)(const CompetitorCard = styled\.div`)', r'\1const CompetitorKeywordCard = styled.div`'),
    (r'(2012:)(const CompetitorCardHeader = styled\.div`)', r'\1const CompetitorKeywordCardHeader = styled.div`'),
    (r'(2019:)(const CompetitorName = styled\.h4`)', r'\1const CompetitorKeywordName = styled.h4`'),
    (r'(2025:)(const CompetitorBadge = styled\.span`)', r'\1const CompetitorKeywordBadge = styled.span`'),
    (r'(2038:)(const CompetitorMetric = styled\.div`)', r'\1const CompetitorKeywordMetric = styled.div`'),
    (r'(2049:)(const CompetitorMetricLabel = styled\.span`)', r'\1const CompetitorKeywordMetricLabel = styled.span`'),
    (r'(2054:)(const CompetitorMetricValue = styled\.span`)', r'\1const CompetitorKeywordMetricValue = styled.span`'),
]

# Apply replacements
content_new = content
for pattern, replacement in replacements:
    content_new = re.sub(pattern, replacement, content_new, flags=re.MULTILINE)

# Also replace in JSX usage
content_new = re.sub(r'<CompetitorCard([^>]*key={index}', '<CompetitorKeywordCard\1', content_new)
content_new = re.sub(r'</CompetitorCard>', '</CompetitorKeywordCard>', content_new)
content_new = re.sub(r'<CompetitorCardHeader>', '<CompetitorKeywordCardHeader>', content_new)
content_new = re.sub(r'</CompetitorCardHeader>', '</CompetitorKeywordCardHeader>', content_new)
content_new = re.sub(r'<CompetitorName>', '<CompetitorKeywordName>', content_new)
content_new = re.sub(r'</CompetitorName>', '</CompetitorKeywordName>', content_new)
content_new = re.sub(r'<CompetitorBadge([^>]*>)', '<CompetitorKeywordBadge\1', content_new)
content_new = re.sub(r'</CompetitorBadge>', '</CompetitorKeywordBadge>', content_new)
content_new = re.sub(r'<CompetitorMetric>', '<CompetitorKeywordMetric>', content_new)
content_new = re.sub(r'</CompetitorMetric>', '</CompetitorKeywordMetric>', content_new)
content_new = re.sub(r'<CompetitorMetricLabel([^>]*)>', '<CompetitorKeywordMetricLabel\1>', content_new)
content_new = re.sub(r'</CompetitorMetricLabel>', '</CompetitorKeywordMetricLabel>', content_new)
content_new = re.sub(r'<CompetitorMetricValue([^>]*)>', '<CompetitorKeywordMetricValue\1>', content_new)
content_new = re.sub(r'</CompetitorMetricValue>', '</CompetitorKeywordMetricValue>', content_new)

# Write back
with open('frontend/src/pages/StrategicDashboard.jsx', 'w', encoding='utf-8') as f:
    f.write(content_new)

print("Components renamed successfully!")
